examples/spisd/device.c

```diff
@@ -68,7 +69,7 @@ static struct Interrupt *remove_int;
 static struct IOStdReq *change_int;
 
 char device_name[] = "spisd.device";
-char id_string[] = "spisd 2.0 (19 July 2021)";
+char id_string[] = "spisd.device 2.0 (19-Jul-2021)";
```

Comment:
Should the date and number be bumped here?

Reply:
Good point.

To give some background why I touched this line: I wanted it to work properly with the version command so it does not give a different version depending on if it is run before the device has been loaded into memory or after. Also when at it make naming consistent and format the date string so the version command understands it.

I did contemplate bumping it when changing it at first, but thought that ideally the maintainer should do it when he decides to tag the repo and do a release, as there could be more than one active pull request at a time and touching that line would cause an unnecessary conflict.

Anyway, I guess I could as well do it as there are no other pull requests active and I have already touched the line.

Reply:
Btw, noticed that while the version command understands date format with the month name (the three letter variant), it still shows the date part as the other info when you run `version FULL` on it:
```
version FULL spisd.device 
spisd.device 2.0 (2021-07-19)
(19-Jul-2021)
```

So that format is not suitable after all.

Reply:
What do you think?
```8.Ram Disk:> version DEVS:spisd.device 
spisd.device 2.2
8.Ram Disk:> version FULL DEVS:spisd.device 
spisd.device 2.2 (2023-03-05)
by Niklas Ekström```

Btw, I noticed that the version command ignores the revision part of the version string and only uses the version number from the romtag if you load it in memory and reference it by resident name instead full path:
```
8.Ram Disk:> version spisd.device 
spisd.device 2.2
8.Ram Disk:> mount sd0: 
8.Ram Disk:> version spisd.device 
spisd.device 2.0
8.Ram Disk:> version devs:spisd.device 
spisd.device 2.2
8.Ram Disk:> version `which version` 
version 47.3
```

would say this is a bug in the version command - it does parse the version string, else it would not be aware of the date. Using the 3.2.1 version command here.

Reply:
Hi Patrik! I'm starting to look at your pr now.

Overall it looks like a fantastic addition (but I will look through it in more detail).

I looked through the AmigaOS source code and how the ID string is typically formatted. The most common variant is: "audio 36.9 (27.5.92)\n\r"

Another option that occurs is: "dos 35.7 (9 Feb 1990)\n\r"

I suggest changing the id string here to: "spisd 2.2 (5.3.23)\n\r"

Reply:
Perhaps something similar to this can be done in VBCC too? (I haven't checked)
https://github.com/jbilander/SimpleDevice/blob/master/device.c#L16
I believe it will make things easier when bumping version and revision and also avoid missing to update Version in the romtag too.

Reply:
Sounds like a great idea, can have a stab at it later this evening.

Reply:
@idrougge: There was no bug in the version command, I had missed this line of code when attempting to fix the version:
https://github.com/niklasekstrom/amiga-par-to-spi-adapter/blob/master/examples/spisd/device.c#L268

@niklasekstrom: I changed the version string to "spisd.device 2.2 (5.3.2023)"
- 2023 as there is no version of the Version command that it causes issues for and it at least makes it obvious which field is the year. To give some more details:
    - The 3.1 and below version command stops parsing correctly after 1995 or so and it doesn't matter if 23 or 2023 is used.
    - The 1.3 version command does not care about the idString in the romtag, it actually seems to load the device and look at the struct Library version and revision.
- I did not add an ending \n\r as I have never experienced or read that it is required anywhere. This device is a great example of this - it has managed perfectly without them so far. I think it is the classic phenomena that things nobody know what they do keep being replicated.

@jbilander: Does this look better?

Reply:
No single library/device/resource in the AmigaOS source has the file extension (.device) in the id string. I suggest you do the same here.

Reply:
Interesting, I assumed so as they all present as somename.device ver.rev when using the version command on them.

I see, they use xyz.device in the struct Library name, but indeed you are correct that the idString is without .device. You learn something new every day, cheers :).

Also found one motivation to use \n\r: Scout does not show the idString unless it ends with that.

Pushed an update.

Reply:
Quite complicated this how the version command in 3.1+ (I assume 2.0+, but have not tested) work libraries/devices.
If the device is not loaded:
- Name is sourced from romtag name string
- Version is sourced from romtag version number number
- Revision is sourced from revision part of romtag idString string
- Date is sourced from date part of romtag idString string

If the device is loaded/initialized:
- Name is sourced from library name string
- Version is sourced from library version number
- Revision is sourced from library revision number
- Date is sourced from date part of library idString string

I must admin I confused the idString with a $VER version string, which interacts way more straight forward way, did not intend to come of as ignorant.

What triggered fiddling with the version part of the code was that when versioning the original device it said spisd.device 1.0 and after it was loaded it said spisd.device 2.0, while the git tag said 2.1 and I figured that if this pr got merged, it would be hard for users to tell if they had the old or new version if there would be any issues.

Reply:
All good! I think fixing the version-thingy seems like a great change, and you have done a thorough job figuring out how it works.

Reply:
Cheers :). Pushed a formatting update as I noticed I had failed to keep in line with the repo formatting for much of the version change and one line in device.c.